vividus/src/main/java/org/vividus/bdd/steps/AssertSteps.java

```diff
@@ -40,7 +40,7 @@ public void verifyIfAssertionsPassed()
                 .skip(1)
                 .map(RunningStory::getRunningScenario)
                 .anyMatch(Objects::nonNull);
-        if (bddRunContext.getRunningStory() == bddRunContext.getRootRunningStory()
+        if (bddRunContext.getRunningStory().equals(bddRunContext.getRootRunningStory())
```

Comment:
== is expected, since we want to check the objects are same

Reply:
It's checked this way transitively